Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -19,6 +19,64 @@ Param(
     [string] $get = ""
 )
 
+function Get-ChangedFiles($token) {
+    $headers = @{             
+        "Authorization" = "token $token"
+        "Accept" = "application/vnd.github.baptiste-preview+json"
+    }
+    $ghEvent = Get-Content $ENV:GITHUB_EVENT_PATH -encoding UTF8 | ConvertFrom-Json
+
+    $url = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/compare/$($ghEvent.pull_request.base.sha)...$($ENV:GITHUB_SHA)"
+
+    $response = InvokeWebRequest -Headers $headers -Uri $url | ConvertFrom-Json
+    $filesChanged = @($response.files | ForEach-Object { $_.filename })
+
+    return $filesChanged
+}
+
+function Get-ProjectsToBuild($settings, $projects, $baseFolder, $token) {
+    if ($settings.alwaysBuildAllProjects) {
+        Write-Host "Building all projects because alwaysBuildAllProjects is set to true"
+        return $projects
+    } elseif ($ENV:GITHUB_EVENT_NAME -notin @("pull_request_target", "pull_request")) {
+        Write-Host "Building all projects because this is not a pull request"
+        return $projects
+    }
+    else {
+        $filesChanged = @(Get-ChangedFiles -token $token)
+        if ($filesChanged.Count -eq 0) {
+            Write-Host "Building all projects"
```

Comment:
Building all projects, because...? 😕

Reply:
Now that we only do incremental builds for PRs, this shouldn't be needed